test/Proxy.js

```diff
@@ -0,0 +1,138 @@
+    const auctioneer = await dx.auctioneer.call()
+    assert.strictEqual(auctioneer, acc, 'account should be DutchExchange contract auctioneer')
+  }
+
+  const assertIsNotAuctioneer = async (acc) => {
+    const auctioneer = await dx.auctioneer.call()
+    assert.notStrictEqual(auctioneer, acc, 'account should not be DutchExchange contract auctioneer')
+  }
+
+  it('DutchExchange is initialized and params are set', async () => {
+    const isInitialised = await dx.isInitialised.call()
+    assert.isTrue(isInitialised, 'DutchExchange should be initialized')
+
+    const params = await getExchangeParams()
+    assert.isTrue(Object.values(params).every(param => !!+param), 'No zero-initialized parameters')
+  })
+
+  it('masterCopy can\'t be updated before masterCopyCountdown was started', async () => {
+    assertIsAuctioneer(master)
+    log('calling dx.updateMasterCopy() as auctioneer')
+    await assertRejects(dx.updateMasterCopy({ from: master }), 'should reject as startMasterCopyCountdown wasn\'t yet called')
+    log('tx was rejected')
+  })
+
+  it('not auctioneer can\'t call startMasterCopyCountdown', async () => {
+    assertIsNotAuctioneer(seller1)
+    log('calling dx.startMasterCopyCountdown() as not auctioneer')
+    await assertRejects(dx.startMasterCopyCountdown(dxNew.address, { from: seller1 }), 'should reject as caller isn\'t the auctioneer')
+    log('tx was rejected')
+  })
+
+  it('can\'t call startMasterCopyCountdown with zero dx address', async () => {
+    assertIsAuctioneer(master)
+    log('calling dx.startMasterCopyCountdown() with dx address == 0')
+    await assertRejects(dx.startMasterCopyCountdown(0, { from: master }), 'should reject as caller isn\'t the auctioneer')
+    log('tx was rejected')
+  })
+
+  it('auctioneer can call startMasterCopyCountdown', async () => {
+    assertIsAuctioneer(master)
+    log('calling dx.startMasterCopyCountdown() as auctioneer with valid dx address')
+    await dx.startMasterCopyCountdown(dxNew.address, { from: master })
+  })
+
+  it('auctioneer can\'t update masterCopy before time limit', async () => {
+    assertIsAuctioneer(master)
+    log('calling dx.updateMasterCopy() as auctioneer before time limit')
+    await assertRejects(dx.updateMasterCopy({ from: master }), 'should reject as time hasn\t passed')
+    log('tx was rejected')
+  })
+
+  it('not auctioneer can\'t update masterCopy', async () => {
+    await wait(60 * 60 * 24 * 30)
+    assertIsNotAuctioneer(seller1)
+    log('calling dx.updateMasterCopy() as not auctioneer after time limit')
+    await assertRejects(dx.updateMasterCopy({ from: seller1 }), 'should reject as caller isn\'t the auctioneer')
+    log('tx was rejected')
+  })
+
+  it('auctioneer can update masterCopy after time limit', async () => {
+    assertIsAuctioneer(master)
+    const params1 = await getExchangeParams()
+
+    log('calling dx.updateMasterCopy() as auctioneer after time limit')
+    await dx.updateMasterCopy({ from: master })
+
```

Comment:
I would add here one small assert.equal(dxNew.address,await dx.masterCopy.call())

but probably you have not done this because masteCopy is not public. Should we make it public in general?

Anyways this is just details. Good testing.